lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -445,6 +434,10 @@ ExhaustiveSynthesis::synthesize(SMTLIBSolver *SMTSolver,
   // find the valid one
   int Unsat = 0;
   int GuessIndex = -1;
+
+  std::vector<Inst *> Vars;
+  findVars(LHS, Vars);
```

Comment:
what's the reason for moving this code earlier?

Reply:
This code is inside the again: label and evaluated on each constant guess try. I am only moving this before the again: label

Reply:
aha, great catch, thanks!